watchtower-plugin/src/retrier.rs

```diff
@@ -261,23 +369,25 @@ impl Retrier {
                 .await
                 .map_err(|e| {
                     log::debug!("Cannot renew registration with tower. Error: {:?}", e);
-                    Error::permanent("Cannot renew registration with tower")
+                    Error::permanent(RetryError::Register(
+                        "Cannot renew registration with tower".to_owned(),
+                    ))
```

Comment:
Ok so permanent errors are the one requiring manual retry (right?),
What if the registeration failed because for a split second the internet connection was down, we should try again IMO.

Reply:
Yeah, I was also thining that `Register` errors (or a subset of them) may need to be transitory instead of permanent. I think there's a `DISCUSS` tag in the `is_permanent` implementation. Would love to hear you opinion on which would you consider permanent and which will you consider transient